src/main/java/bio/terra/service/auth/iam/IamService.java

```diff
@@ -440,6 +440,15 @@ public String createGroup(String groupName) {
     return callProvider(() -> iamProvider.createGroup(tdrSaAccessToken, groupName));
   }
 
+  /**
+   * @param groupName Firecloud managed group to retrieve as the TDR SA
+   * @return the email for the retrieved group
+   */
+  public String getGroup(String groupName) {
+    String tdrSaAccessToken = googleCredentialsService.getApplicationDefaultAccessToken(SCOPES);
+    return callProvider(() -> iamProvider.getGroup(tdrSaAccessToken, groupName));
+  }
```

Comment:
That's good to note - this is creating the group as the TDR SA (as opposed to the current user). So, any group created by TDR should be accessible to the SA.

Reply:
Is this correct, to generate the group using the TDR SA? We do have the requesting user token on hand, should that be used instead? I'm not sure what the implications are for not using the user's SA.